module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -542,6 +542,42 @@ protected function fetchUserWithCql($query)
         return count($json->users) === 1 ? $json->users[0] : null;
     }
 
+    /**
+     * Helper function to retrieve paged results from FOLIO API
+     *
+     * @param string $responseKey Key containing values to collect in response
+     * @param string $interface   FOLIO api interface to call
+     * @param array  $query       CQL query
+     *
+     * @return array
+     */
+    protected function getPagedResults($responseKey, $interface, $query = [])
+    {
+        $fullResponse = [];
+        // set artificially low to test
+        $limit = 1000;
+        $offset = 0;
+
+        do {
+            $combinedQuery = compact('query', 'offset', 'limit');
```

Comment:
Do you really want to `compact` query here? That's going to create an array with a key of `'query'` and a value of the incoming `$query` parameter, which seems like it may not have the desired effect. Do you actually want to `array_merge` `$query` with the compacted `$offset` and `$limit`?

Reply:
Thank you. Really appreciate the review for things like this. I'm learning php.